core/src/main/scala/kafka/server/QuotaFactory.scala

```diff
@@ -0,0 +1,62 @@
+package kafka.server
+
+import kafka.common.TopicAndPartition
+import org.apache.kafka.common.metrics.Metrics
+import org.apache.kafka.common.utils.SystemTime
+
+import scala.collection.Map
+
+object QuotaType  {
+  case object Fetch extends QuotaType
+  case object Produce extends QuotaType
+  case object LeaderReplication extends QuotaType
+  case object  FollowerReplication extends QuotaType
+}
+sealed trait QuotaType
+
+object QuotaFactory {
+
+
+  object UnbreakableQuota extends ReadOnlyQuota {
+    override def bound(): Int = Int.MaxValue
+
+    override def isThrottled(topicAndPartition: TopicAndPartition): Boolean = false
+
+    override def isQuotaExceededBy(bytes: Int): Boolean = false
+  }
+
+  case class QuotaManagers(client: Map[QuotaType, ClientQuotaManager], leaderReplication: ReplicationQuotaManager, followerReplication: ReplicationQuotaManager)
```

Comment:
Given that the containing class is `QuotaManagers`, it seems like it may be OK to not repeat `QuotaManager` for each field. If we do rename it as per Jun's suggestion, then we probably should rename `client` too.